catalog/core/catalog-core-transform-impl/src/main/java/org/codice/ddf/catalog/transform/impl/TransformImpl.java

```diff
@@ -0,0 +1,369 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.transform.impl;
+
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardCreationException;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.QueryResponseTransformer;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.activation.MimeType;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.codice.ddf.catalog.locator.TransformerLocator;
+import org.codice.ddf.catalog.transform.MultiInputTransformer;
+import org.codice.ddf.catalog.transform.MultiMetacardTransformer;
+import org.codice.ddf.catalog.transform.Transform;
+import org.codice.ddf.platform.util.TemporaryFileBackedOutputStream;
+import org.codice.ddf.platform.util.uuidgenerator.UuidGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TransformImpl implements Transform {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TransformImpl.class);
+
+  private TransformerLocator transformerLocator;
+
+  private UuidGenerator uuidGenerator;
+
+  public TransformImpl(TransformerLocator transformerLocator, UuidGenerator uuidGenerator) {
+    this.transformerLocator = transformerLocator;
+    this.uuidGenerator = uuidGenerator;
+  }
+
+  @Override
+  public List<Metacard> transform(
+      MimeType mimeType,
+      Supplier<String> idSupplier,
+      String fileName,
+      File inputFile,
+      String transformerId,
+      Map<String, ? extends Serializable> transformerArguments)
+      throws MetacardCreationException {
+    try (InputStream transformerStream =
+        com.google.common.io.Files.asByteSource(inputFile).openStream()) {
```

Comment:
do we need to use the full package name here?

Reply:
fixed